components/settings/settings.cpp

```diff
@@ -52,96 +73,224 @@ namespace Settings
     {
-        Container properties;
-        const boost::property_tree::ptree& pt = mUserPropertyTree;
-        boost::property_tree::ptree::const_assoc_iterator assocIterator = pt.find(section);
+        CSimpleIniCaseA::TNamesDepend loadedKeys;
+        mImpl->mUserIni.GetAllKeys(section.c_str(), loadedKeys);
 
-        if (assocIterator != pt.not_found())
+        Container properties;
+        for (const CSimpleIniCaseA::Entry key : loadedKeys)
         {
-            for (boost::property_tree::ptree::const_iterator key = assocIterator->second.begin(); key != assocIterator->second.end(); ++key)
-            {
-                properties.push_back(key->first);
-            }
+            properties.push_back(key.pItem);
         }
-
         return properties;
     }
 
     bool Settings::isSectionExists(const std::string& section)
     {
-        const boost::property_tree::ptree& pt = mUserPropertyTree;
-        boost::property_tree::ptree::const_assoc_iterator assocIterator = pt.find(section);
-        return assocIterator != pt.not_found();
+        const int sectionSize = mImpl->mUserIni.GetSectionSize(section.c_str());
+        return -1 < sectionSize;
+    }
+
+    template <> std::string Settings::get<std::string>(const std::string& section, const std::string& name, std::string defaultValue)
+    {
+        if (!mImpl->isKeyExists(section, name))
+        {
+            return mImpl->mDefaultsIni.GetValue(section.c_str(), name.c_str(), defaultValue.c_str());
+        }
+
+        return mImpl->mUserIni.GetValue(section.c_str(), name.c_str(), defaultValue.c_str());
+    }
+
+    template <> int64_t Settings::get<int64_t>(const std::string& section, const std::string& name, int64_t defaultValue)
+    {
+        /// Do not rely on CSimpleIniCaseA::GetLongValue(), as it has limited range.
+        /// Hexadecimal numbers are auto-detected by std::stoll().
+        const std::string textValue = get<std::string>(section, name, std::to_string(defaultValue));
+        return std::stoll(textValue, nullptr, 0);
+    }
+
+    template <> int32_t Settings::get<int32_t>(const std::string& section, const std::string& name, int32_t defaultValue)
+    {
+        return get<int64_t>(section, name, defaultValue);
+    }
+
+    template <> int16_t Settings::get<int16_t>(const std::string& section, const std::string& name, int16_t defaultValue)
+    {
+        return get<int64_t>(section, name, defaultValue);
+    }
+
+    template <> int8_t Settings::get<int8_t>(const std::string& section, const std::string& name, int8_t defaultValue)
+    {
+        return get<int64_t>(section, name, defaultValue);
+    }
+
+    template <> uint64_t Settings::get<uint64_t>(const std::string& section, const std::string& name, uint64_t defaultValue)
+    {
+        const std::string textValue = get<std::string>(section, name, std::to_string(defaultValue));
+        return std::stoull(textValue, nullptr, 0);
+    }
+
+#ifdef __APPLE__
```

Comment:
Should it be checking for clang, instead of apple?
I think a better solution might be to change https://github.com/balintkiss501/freeablo/blob/388_replace_boost_ini_parser/components/diabloexe/diabloexe.cpp#L150 to use uint34_t instead, so this overload isn't needed.

Reply:
> Should it be checking for clang, instead of apple?
Yes, `ifdef __clang__` would be definitely better.

But I'd be happier without this workaround and with the change in that `diabloexe.cpp` you mentioned. Looking back in the failed build log, this was the only function including that `auto` keyword that caused trouble.

Reply:
Osx build failed again, @wheybags do you wish to replace occourences of `auto` with `size_t` in `diabloexe.cpp`?

Reply:
Can We just add an overload for size_t that forwards to uint64_t?

Reply:
I set Clang on my machine just for this and to me, it seems like

```cpp
#ifdef __clang__
template <> size_t Settings::get<size_t>(const std::string& section, const std::string& name, size_t defaultValue)

    return get<uint64_t>(section, name, defaultValue);
}
#endif
```

compiles on both GCC and Clang.

Reply:
I'm OK with your commit.